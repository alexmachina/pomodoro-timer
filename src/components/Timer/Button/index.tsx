import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const StyledButton = styled.button`
  background-color: #ff4785;
  color: #fff;
  outline: none;
  border: none;
  border-radius: 4px;
  padding: 4px;
  width: 80px;

  &:hover {
    background-color: #ee2285;
  }
`;

interface PlayButtonProps {
  onClick: (e: React.MouseEvent) => void;
  children: React.ReactNode;
}

const Button: React.FC<PlayButtonProps> = ({ onClick, children }) => {
  return <StyledButton onClick={onClick}>{children}</StyledButton>;
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
