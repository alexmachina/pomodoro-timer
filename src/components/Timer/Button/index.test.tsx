import React from "react";
import { render } from "@testing-library/react";
import Button from "./index";

test("Button", () => {
  const onClick = jest.fn();
  const { container } = render(<Button onClick={onClick}>Play</Button>);

  expect(container).toMatchSnapshot();
});
