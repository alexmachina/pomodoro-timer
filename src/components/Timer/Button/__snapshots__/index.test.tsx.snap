// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One + 1 1`] = `
<div>
  <button
    class="sc-AxjAm lhBPoE"
  >
    Play
  </button>
</div>
`;

exports[`PlayButton 1`] = `
<div>
  <button
    class="sc-AxjAm lhBPoE"
  >
    Play
  </button>
</div>
`;
